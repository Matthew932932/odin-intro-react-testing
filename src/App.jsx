import { useState } from 'react'
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'

// App.jsx



const App = () => {
  const [heading, setHeading] = useState("Magnificent Monkeys");

  const clickHandler = () => {
    setHeading("Radical Rhinos");
  };

  return (
    <>
      <button type="button" onClick={clickHandler}>
        Click Me 1
      </button>
      <h1>{heading}</h1>
    </>
  );
};

export default App;


