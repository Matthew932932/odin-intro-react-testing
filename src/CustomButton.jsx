// CustomButton.jsx

const CustomButton = ({ onClick }) => {
    return (
      <button onClick={onClick}>Click me 2</button> 
    );
  };
  
  export default CustomButton;
  