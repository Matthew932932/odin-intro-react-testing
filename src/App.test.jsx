// App.test.jsx

import { vi, describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";
import CustomButton from "./CustomButton";

describe("App component", () => {
  it("renders magnificent monkeys", () => {
    // since screen does not have the container property, we'll destructure render to obtain a container for this test
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders radical rhinos after button click", async () => {
    const user = userEvent.setup();

    const { container } = render(<App />);
    const button = screen.getByRole("button", { name: "Click Me 1" });

    await user.click(button);


    expect(container).toMatchSnapshot();
    //expect(screen.getByRole("heading").textContent).toMatch(/radical rhinos/i);
  });
});

describe("CustomButton", () => {
  it("should render a button with the text 'Click me'", () => {
    render(<CustomButton onClick={() => {}} />);

    const button = screen.getByRole("button", { name: "Click me 2" });

    expect(button).toBeInTheDocument();
  });
  
  it("should call the onClick function when clicked", async () => {
    const onClick = vi.fn();
    const user = userEvent.setup()
    render(<CustomButton onClick={onClick} />);

    const button = screen.getByRole("button", { name: "Click me 2" });

    await user.click(button);

    expect(onClick).toHaveBeenCalled();
  });

  it("should not call the onClick function when it isn't clicked", async () => {
    const onClick = vi.fn();
    render(<CustomButton onClick={onClick} />);

    expect(onClick).not.toHaveBeenCalled();
  });
});
