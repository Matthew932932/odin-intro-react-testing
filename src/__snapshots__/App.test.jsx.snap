// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders magnificent monkeys 1`] = `
<div>
  <button
    type="button"
  >
    Click Me 1
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
</div>
`;

exports[`App component > renders radical rhinos after button click 1`] = `
<div>
  <button
    type="button"
  >
    Click Me 1
  </button>
  <h1>
    Radical Rhinos
  </h1>
</div>
`;
